spec2vec/Spec2Vec.py

```diff
@@ -132,11 +162,26 @@ def matrix(self, references: List[SpectrumDocument], queries: List[SpectrumDocum
             query_vectors = numpy.empty((n_cols, self.vector_size), dtype="float")
             for index_query, query in enumerate(tqdm(queries, desc='Calculating vectors of query spectrums',
                                                      disable=self.disable_progress_bar)):
-                query_vectors[index_query, 0:self.vector_size] = calc_vector(self.model,
-                                                                             query,
-                                                                             self.intensity_weighting_power,
-                                                                             self.allowed_missing_percentage)
+                query_vectors[index_query, 0:self.vector_size] = self._calculate_embedding(query)
 
         spec2vec_similarity = cosine_similarity_matrix(reference_vectors, query_vectors)
 
         return spec2vec_similarity
+
+    @staticmethod
+    def _get_word_decimals(model):
+        """Read the decimal rounding that was used to train the model"""
+        word_regex = r"[a-z]{4}@[0-9]{1,5}."
+        example_word = next(iter(model.wv.vocab))
+        return len(re.split(word_regex, example_word)[-1])
+
+    def _calculate_embedding(self, spectrum_in: Union[SpectrumDocument, SpectrumType]):
+        """Generate Spec2Vec embedding vectors from input spectrum (or SpectrumDocument)"""
+        if isinstance(spectrum_in, Spectrum):
+            spectrum_in = SpectrumDocument(spectrum_in, n_decimals=self.n_decimals)
+        assert spectrum_in.n_decimals == self.n_decimals, \
+            "Decimal rounding of input data does not agree with model vocabulary."
```

Comment:
This can only be False when spectrum_in is already a SpectrumDocument. So assert could be skipped when spectrum_in is a Spectrum
```suggestion
        else:
            assert spectrum_in.n_decimals == self.n_decimals, \
                "Decimal rounding of input data does not agree with model vocabulary."
```

Reply:
Good point. Thanks.